tests/monitoring/monitoring.go

```diff
@@ -0,0 +1,143 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package monitoring
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"kubevirt.io/kubevirt/tests"
+
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"kubevirt.io/kubevirt/tests/flags"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"kubevirt.io/client-go/kubecli"
+)
+
+const (
+	virtOperatorDeploymentName = "virt-operator"
+)
+
+var _ = Describe("[Serial][sig-monitoring]Prometheus Alerts", func() {
+
+	var err error
+	var virtClient kubecli.KubevirtClient
+	var scales = make(map[string]*autoscalingv1.Scale, 1)
+
+	backupScale := func(operatorName string) {
+		Eventually(func() error {
+			virtOperatorCurrentScale, err := virtClient.AppsV1().Deployments(flags.KubeVirtInstallNamespace).GetScale(context.TODO(), operatorName, metav1.GetOptions{})
+			if err == nil {
+				scales[operatorName] = virtOperatorCurrentScale
+			}
+			return err
+		}, 30*time.Second, 1*time.Second).Should(BeNil())
+	}
+
+	revertScale := func(operatorName string) {
+		revert := scales[operatorName].DeepCopy()
+		revert.ResourceVersion = ""
+		Eventually(func() error {
+			_, err = virtClient.
+				AppsV1().
+				Deployments(flags.KubeVirtInstallNamespace).
+				UpdateScale(context.TODO(), operatorName, revert, metav1.UpdateOptions{})
+			return err
+		}, 30*time.Second, 1*time.Second).Should(BeNil())
+	}
+
+	updateScale := func(operatorName string, replicas int32) {
+		scale := scales[operatorName].DeepCopy()
+		scale.Spec.Replicas = replicas
+		Eventually(func() error {
+			_, err = virtClient.
+				AppsV1().
+				Deployments(flags.KubeVirtInstallNamespace).
+				UpdateScale(context.TODO(), operatorName, scale, metav1.UpdateOptions{})
+			return err
+		}, 30*time.Second, 1*time.Second).Should(BeNil())
+	}
+
+	verifyAlertExist := func(alertName string) {
+		Eventually(func() error {
+			alerts, err := getAlerts(virtClient)
+			if err != nil {
+				return err
+			}
+			for _, alert := range alerts {
+				if string(alert.Labels["alertname"]) == alertName {
+					return nil
+				}
+			}
+			return fmt.Errorf("alert doesn't exist: %v", alertName)
+		}, 120*time.Second, 1*time.Second).Should(BeNil())
+
+	}
+
+	waitUntilThereIsNoAlert := func() {
+		Eventually(func() error {
+			alerts, err := getAlerts(virtClient)
+			if err != nil {
+				return err
+			}
+			if len(alerts) == 0 {
+				return nil
+			}
+			return fmt.Errorf("some alerts exist: %v", alerts)
+		}, 120*time.Second, 1*time.Second).Should(BeNil())
+
+	}
+
+	BeforeEach(func() {
+		virtClient, err = kubecli.GetKubevirtClient()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(virtClient).ToNot(BeNil())
+
+		tests.SkipIfPrometheusRuleIsNotEnabled(virtClient)
+		tests.BeforeTestCleanup()
+	})
+
+	AfterEach(func() {
+		tests.AfterTestSuitCleanup()
```

Comment:
Why are you calling this explicitly here? This should not be required.

Reply:
removed